optimas/generators/ax/service/base.py

```diff
@@ -171,9 +171,7 @@ def _tell(self, trials: List[Trial]) -> None:
                     current_step = generation_strategy.current_step
                     # Reduce only if there are still Sobol trials left.
                     if current_step.model == Models.SOBOL:
-                        current_step.num_trials -= 1
```

Comment:
This is the actual fix. Just removing this line.

The rest of the changes are simply due to requiring Ax > 0.3.5